File: src/Days 11 to 20/Day 11/kotlin/main.kt
fun main() {
    val a = "abcd"
    val b = "cdabcdab"
    println(findMinRepeats(a, b))
}

fun findMinRepeats(a: String, b: String): Int {
    val lenA = a.length
    val lenB = b.length
    val repeatCount = (lenB + lenA - 1) / lenA

    var repeatedA = a.repeat(repeatCount)

    if (b in repeatedA) {
        return repeatCount
    }

    repeatedA += a

    if (b in repeatedA) {
        return repeatCount + 1
    }

    return -1
}

File: src/Days 11 to 20/Day 11/python/main.py
def min_repeats(a, b):
    repeat_count = -(-len(b) // len(a))  # This is equivalent to math.ceil(len(b) / len(a))
    
    repeated_a = a * repeat_count
    
    if b in repeated_a:
        return repeat_count
    
    repeated_a += a
    if b in repeated_a:
        return repeat_count + 1
    
    return -1

a = "abcd"
b = "cdabcdab"
print(min_repeats(a, b))